internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -744,66 +740,47 @@ func (s *Server) processRelevantRepositoryEvent(
 
 	l.Info().Msg("handling event for repository")
 
-	repoEntity, err := s.fetchRepo(ctx, event.GetRepo())
+	lookByProps, err := properties.NewProperties(map[string]any{
+		properties.PropertyUpstreamID: properties.NumericalValueToUpstreamID(event.GetRepo().GetID()),
+	})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating repository lookup properties: %w", err)
 	}
 
-	hookId, hasHookErr := repoEntity.Properties.GetProperty(ghprop.RepoPropertyHookId).AsInt64()
+	msg := entityMessage.NewEntityRefreshAndDoMessage().
+		WithEntity(pb.Entity_ENTITY_REPOSITORIES, lookByProps).
+		WithProviderImplementsHint(string(db.ProviderTypeGithub))
+
 	// This only makes sense for "meta" event type
-	if event.GetHookID() != 0 && hasHookErr == nil {
+	if event.GetHookID() != 0 {
 		// Check if the payload webhook ID matches the one we
 		// have stored in the DB for this repository
-		if event.GetHookID() != hookId {
-			// This means we got a deleted event for a
-			// webhook ID that doesn't correspond to the
-			// one we have stored in the DB.
-			return nil, newErrNotHandled("meta event with action %s not handled, hook ID %d does not match stored webhook ID %d",
-				event.GetAction(),
-				event.GetHookID(),
-				hookId,
-			)
+		// If not, this means we got a deleted event for a
+		// webhook ID that doesn't correspond to the
+		// one we have stored in the DB.
+		matchHookProps, err := properties.NewProperties(map[string]any{
```

Comment:
This codepath was not easy to test. In the end I resorted to a trick and:
 - registered a repo
 - changed the hookID value in the DB
 - deleted the webhook
 - made sure the message is ignored